ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestExecutor.java

```diff
@@ -132,6 +136,23 @@ public static TestExecutor create(
     );
   }
 
+  // These internal topics names do not need to be verified by test cases
+  private static final Set<Pattern> SKIP_INTERNAL_TOPIC_NAMES_PATTERNS = ImmutableSet.of(
```

Comment:
I don't think this belongs in `TestExecutor`. It should be up to the `TopicInfoCache` to say that topic info will not be cached for certain topics. How about we move this into `TopicInfoCache` and update `TopicInfoCache.get()` to return `Optional<TopicInfo>` where an empty optional will be returned if there's no info?

Reply:
The cache is not the problem, but when getting the schema/serde info of the topic. Caching is secondary. I'm not sure about returning Optional for internal topics that we don't want. The `TopicInfoCache.get()` contract is to return info about a topic. If there is no info about a topic, the `get()` already throws an exception:
```
throw new TestFrameworkException("No information found for topic '"
            + topicName + "'. Available topics: " + cache.asMap().keySet());
```

So we'll have now an `Optional.empty()` and an exception that mean the same thing. If I return an Optional, then the caller needs to check if it is optional, then throw an exception. 

I agree the list of internal topics don't look that they fit in the TestExecutor, but the TestExecutor is the one that has a list of topics, so IMHO we should filter them out there.